src/analysis/inner_product_space/projection.lean

```diff
@@ -862,6 +885,108 @@ lemma finrank_orthogonal_span_singleton {n : ℕ} [_i : fact (finrank 𝕜 E = n
   finrank 𝕜 (𝕜 ∙ v)ᗮ = n :=
 submodule.finrank_add_finrank_orthogonal' $ by simp [finrank_span_singleton hv, _i.elim, add_comm]
 
+/-- An element `φ` of the orthogonal group of `F` can be factored as a product of reflections, and
+specifically at most as many reflections as the dimension of the complement of the fixed subspace
+of `φ`. -/
+lemma linear_isometry_equiv.reflections_generate_dim_aux [finite_dimensional ℝ F] {n : ℕ}
+  (φ : F ≃ₗᵢ[ℝ] F)
+  (hn : finrank ℝ (continuous_linear_map.id ℝ F - φ.to_continuous_linear_equiv).kerᗮ ≤ n) :
+  ∃ l : list F, l.length ≤ n ∧ φ = (l.map (λ v, reflection (ℝ ∙ v)ᗮ)).prod :=
```

Comment:
Is it worth showing this list is `nodup`?

Reply:
I'd rather not, I don't see an obvious application and I like the conciseness of the current statement.

Reply:
I'm no longer certain it's true anyway!

Reply:
It is true (it's a byproduct of my proof that each vector belongs to the orthogonal complement of all previously-found vectors), but I don't feel like recording it :)